pyugrid/read_netcdf.py

```diff
@@ -150,9 +150,11 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
                 raise ValueError(msg(defs['role']))
             continue
         except KeyError:
-            msg = ("File must include {} variables for {} "
-                   "named in mesh variable.").format
-            raise ValueError(msg(coord_names, defs['role']))
+            if defs['required']:
+                msg = ("File must include {} variables for {} "
+                       "named in mesh variable.").format
+                raise ValueError(msg(coord_names, defs['role']))
+            continue
```

Comment:
> doesn't make for too much code duplication.

:+1: 

> This al started as a hack to get something to work, then I refactored it to be a bit cleaner and more of a compliance checker, but it's by no means done :-)

I get it.  I feel like all my projects are like that :wink: 

>  with the user specifying extra info

Yeah, that should be the priority in the compliance checking.  To inform users of that is missing and to feed what was found back into the pyugrid object creation.